### PR TITLE
Improve types for `sendStateEvent`

### PR DESCRIPTION
**Pull request by t3chguy**
Requires https://github.com/matrix-org/matrix-js-sdk/pull/4105
Requires https://github.com/matrix-org/matrix-react-sdk/pull/12350

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -143,6 +143,7 @@ module.exports = {
                             "!matrix-js-sdk/src/models/read-receipt",
                             "!matrix-js-sdk/src/models/relations-container",
                             "!matrix-js-sdk/src/models/related-relations",
+                            "!matrix-js-sdk/src/models/invites-ignorer",
```
Comment (richvdh) [inline .eslintrc.js:146]: surely we shouldn't be adding new exclusions here, unless we're deliberately defining new entrypoints for the js-sdk?

**Review by richvdh — CHANGES_REQUESTED** [review]
Too much stuff going on at once here. Separate moving stuff between files from changing stuff, please.

File: .eslintrc.js (hunk 1 above)
Comment (t3chguy) [reply .eslintrc.js:146]: The invites-ignorer stuff is an experimental MSC implementation, its not re-exported presumably for the reason of us not wanting to treat it as stable yet

Comment (t3chguy) [reply .eslintrc.js]: Moved to a one-off lint override to not set a bad standard

**Review by richvdh — APPROVED** [review]
seems to be just types and stuff.

**Pull request merged**